pycloudlib/azure/instance.py

```diff
@@ -290,7 +304,82 @@ def add_network_interface(self) -> str:
 
     def remove_network_interface(self, ip_address: str):
         """Remove nic from running instance."""
-        raise NotImplementedError
+        # get ip address id
+        all_ips = list(self._network_client.public_ip_addresses.list_all())
```

Comment:
In all other implementations of this method, `ip_address` is a private IP, this is because a NIC does not necessarily have to have a public IP.

Apart from that, to be consistent with the interface, could we implement it in that way?

Reply:
Both private.

Reply:
Ah, then I am putting up a PR to correct the implementation for the `add_network_interface` since I returned the public one.

Reply:
Mmm, I think we can fix it as part of this PR.

Reply:
Yes, thought it would be a big change, but it is not. I am adding to this.